jupyter-on-gke/jupyterhub.tf

```diff
@@ -26,111 +36,100 @@ provider "helm" {
+# IAP Section: Creates the GKE components
 module "iap_auth" {
   count  = var.add_auth ? 1 : 0
   source = "./iap_module"
 
-  project_id      = var.project_id
-  namespace       = var.namespace
-  service_name    = var.service_name
-  client_id       = var.client_id != "" ? var.client_id : google_iap_client.iap_oauth_client[0].client_id
-  client_secret   = var.client_id != "" ? var.client_secret : google_iap_client.iap_oauth_client[0].secret
-  url_domain_addr = var.url_domain_addr
-  url_domain_name = var.url_domain_name
-
-  depends_on = [
-    helm_release.jupyterhub,
-    kubernetes_namespace.namespace,
-    module.workload_identity_service_account
-  ]
+  project_id              = var.project_id
+  namespace               = var.namespace
+  default_backend_service = var.default_backend_service
+  service_name            = var.service_name
+  client_id               = var.client_id != "" ? var.client_id : google_iap_client.iap_oauth_client[0].client_id
+  client_secret           = var.client_id != "" ? var.client_secret : google_iap_client.iap_oauth_client[0].secret
+  url_domain_addr         = var.url_domain_addr
+  url_domain_name         = var.url_domain_name
+  depends_on              = [google_project_service.project_service]
 }
 
 module "workload_identity_service_account" {
   source = "./service_accounts_module"
 
   project_id      = var.project_id
   namespace       = var.namespace
-  service_account = "jupyter-service-account"
+  service_account = "jupyter-service-account1"
 }
 
 resource "helm_release" "jupyterhub" {
-  name            = "jupyterhub"
-  repository      = "https://jupyterhub.github.io/helm-chart"
-  chart           = "jupyterhub"
-  namespace       = var.namespace
-  cleanup_on_fail = "true"
+  name             = "jupyterhub"
+  repository       = "https://jupyterhub.github.io/helm-chart"
+  chart            = "jupyterhub"
+  namespace        = var.namespace
+  create_namespace = !contains(data.kubernetes_all_namespaces.allns.namespaces, var.namespace)
+  cleanup_on_fail  = "true"
 
   values = [
     templatefile("${path.module}/jupyter_config/config-selfauth.yaml", {
-      service_id          = var.add_auth && data.google_compute_backend_service.jupyter-ingress.generated_id != null ? "${data.google_compute_backend_service.jupyter-ingress.generated_id}" : "no-id-yet"
+      service_id          = var.add_auth && data.google_compute_backend_service.jupyter-ingress.generated_id != null ? data.google_compute_backend_service.jupyter-ingress.generated_id : "no-id-yet"
+      backend_config      = var.service_name
       project_number      = data.google_project.project.number
       authenticator_class = var.add_auth ? "'gcpiapjwtauthenticator.GCPIAPAuthenticator'" : "dummy"
       service_type        = var.add_auth ? "NodePort" : "LoadBalancer"
     })
   ]
+  depends_on = [module.iap_auth]
+}
 
-  depends_on = [
-    kubernetes_namespace.namespace
-  ]
+
+# Need to re-apply: fetch service_id from deployed service
```

Comment:
Is this still true with the helm chart, does it need a re-apply?

Can we also have an updated README with instructions to install?